client/X11/xf_window.c

```diff
@@ -139,19 +139,57 @@ void xf_SendClientEvent(xfContext* xfc, Window window, Atom atom, unsigned int n
 
 void xf_SetWindowFullscreen(xfContext* xfc, xfWindow* window, BOOL fullscreen)
 {
-	if (fullscreen)
+	int i;
+	int startX = xfc->instance->settings->DesktopPosX;
+	int startY = xfc->instance->settings->DesktopPosY;
+
+	window->decorations = xfc->decorations;
+	xf_SetWindowDecorations(xfc, window->handle, window->decorations);
+
+	/* Determine the x,y starting location for the fullscreen window */
+	if (xfc->instance->settings->MonitorCount)
 	{
-		rdpSettings* settings = xfc->settings;
+		/* Initialize startX and startY with reasonable values */
+		startX = xfc->instance->settings->MonitorDefArray[0].x;
+		startY = xfc->instance->settings->MonitorDefArray[0].y;
+
+		/* Search all monitors to find the lowest startX and startY values */
+		for (i=0; i < xfc->instance->settings->MonitorCount; i++)
+		{
+			startX = MIN(startX, xfc->instance->settings->MonitorDefArray[i].x);
+			startY = MIN(startY, xfc->instance->settings->MonitorDefArray[i].y);
+		}
 
-		xf_SetWindowDecorations(xfc, window->handle, FALSE);
+		/* Lastly apply any monitor shift(translation from remote to local coordinate system)
+		 *  to startX and startY values
+		 */
+		startX = startX + xfc->instance->settings->MonitorLocalShiftX;
+		startY = startY + xfc->instance->settings->MonitorLocalShiftY;
+	}
 
-		XMoveResizeWindow(xfc->display, window->handle,
-				settings->DesktopPosX, settings->DesktopPosY, window->width, window->height);
+	XMoveResizeWindow(xfc->display, window->handle, startX, startY, window->width, window->height);
+	XMapRaised(xfc->display, window->handle);
 
-		XMapRaised(xfc->display, window->handle);
+	/* Set the fullscreen state */
+	xf_SendClientEvent(xfc, window->handle, xfc->_NET_WM_STATE, 4,
+				fullscreen ? _NET_WM_STATE_ADD : _NET_WM_STATE_REMOVE,
+				xfc->_NET_WM_STATE_FULLSCREEN, 0, 0);
 
-		window->fullscreen = TRUE;
+	/* Only send monitor bounds if they are valid */
+	if ((xfc->fullscreenMonitors.top >= 0) &&
+			(xfc->fullscreenMonitors.bottom >= 0) &&
+			(xfc->fullscreenMonitors.left >= 0) &&
+			(xfc->fullscreenMonitors.right >= 0))
+	{
+		xf_SendClientEvent(xfc, window->handle, xfc->_NET_WM_FULLSCREEN_MONITORS, 5,
+				xfc->fullscreenMonitors.top,
+				xfc->fullscreenMonitors.bottom,
+				xfc->fullscreenMonitors.left,
+				xfc->fullscreenMonitors.right,
+				1);
 	}
+
+	window->fullscreen = TRUE;
```

Comment:
Yes. It looks like it should. I do not think that variable is actually used by anything, which is probably why it got overlooked and did not cause me any functional problems.